bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/catalog/VOTableCatalog.scala

```diff
@@ -15,28 +16,26 @@ trait VOTableCatalog extends Catalog {
   def host: String
   def decode(vot: VOTable): Seq[Target]
 
-  def find(id: String)(callback: Result => Unit) {
-    actor {
-      val f = callback.safe
-      try {
-
-        val conn = url(id).openConnection
-        conn.setReadTimeout(1000 * 10) // 10 secs?
-
-        // Parse the URL data into a VOTable
-        val vot = VOTable(conn.getInputStream)
-        val ts = decode(vot)
-        if (ts.nonEmpty) f(Success(ts.toList, Nil)) else f(NotFound(id))
-
-      } catch {
-        case e:IOException =>
-          Log.warning("%s: %s(%s)".format(host, e.getClass.getSimpleName, e.getMessage))
-          f(Offline)
-        case t:Throwable =>
-          Log.log(Level.WARNING, "Unexpected trouble looking up %s on %s.".format(id, host), t)
-          f(Error(t))
-      }
+  override def find(id: String)(implicit ex: ExecutionContext): Future[Result] =
+    Future {
```

Comment:
In principle the body of this future should be inside a `blocking { ... }` call, which will let the EC know that it might want to perform the work on a blocking pool. This is implemented with a `ThreadLocal` so it's really sketchy, but at least for documentation I wanted to leave this comment in case we run into a starvation problem. I think it's probably ok as-is.

Reply:
Thanks, it was easy to update regardless so I just did it.